radixdlt-core/radixdlt/src/main/java/com/radixdlt/network/transport/udp/UDPNettyMessageHandler.java

```diff
@@ -41,40 +46,68 @@
 final class UDPNettyMessageHandler extends SimpleChannelInboundHandler<DatagramPacket> {
 	private static final Logger log = LogManager.getLogger();
 
-	private final InboundMessageConsumer messageSink;
+	private final RateLimiter droppedMessagesRateLimiter = RateLimiter.create(1.0);
+	private final RateLimiter logRateLimiter = RateLimiter.create(1.0);
+
+	private final SystemCounters counters;
+	private final int bufferSize;
+
 	private final NatHandler natHandler;
 
+	private final PublishProcessor<Pair<InetSocketAddress, ByteBuf>> rawMessageSink = PublishProcessor.create();
 
-	UDPNettyMessageHandler(NatHandler natHandler, InboundMessageConsumer messageSink) {
-		this.messageSink = messageSink;
+	UDPNettyMessageHandler(SystemCounters counters, int bufferSize, NatHandler natHandler) {
+		this.counters = counters;
+		this.bufferSize = bufferSize;
 		this.natHandler = natHandler;
 	}
 
+	Flowable<InboundMessage> inboundMessageRx() {
+		return rawMessageSink
+			.onBackpressureBuffer(
+				this.bufferSize,
+				() -> {
+					this.counters.increment(SystemCounters.CounterType.NETWORKING_UDP_DROPPED_MESSAGES);
+					if (droppedMessagesRateLimiter.tryAcquire()) {
+						log.warn("UDP msg buffer overflow, dropping msg");
+					}
```

Comment:
Would like to see dropped message logs at trace level if rate limiter limits rate.  Something like:
```
Level logLevel = droppedMessagesRateLimiter.tryAcquire() ? Level.WARN : Level.TRACE;
log.log(logLevel, ...);
```